sound/soc/sof/Makefile

```diff
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
 
 snd-sof-objs := core.o ops.o loader.o ipc.o pcm.o pm.o debug.o topology.o\
-		control.o trace.o utils.o sof-audio.o
+		control.o trace.o utils.o sof-audio.o sof-client.o
```

Comment:
shouldn't sof-client.o be in an separate module? I think I asked this before but I can't recall the answer.

You could have something like 
snd-sof-$(CONFIG_SND_SOC_SOF_CLIENT) := sof-client.o